### PR TITLE
suspend in batch, rather than all at once

### PR DESCRIPTION
**Pull request by AaronCMuller**
@scoen

### CODE REVIEW

File: src/amberdb/graph/AmberGraph.java [hunk 1]
```diff
@@ -250,7 +250,18 @@ public Vertex addVertex(Object id) {
 
     
     public Long suspend() {
-    
+
+        if (getModifiedElementCount() > BIG_COMMIT_THRESHOLD) {
+            log.warn("Graph to be committed exceeds {} elements. Using big suspend to process.", BIG_COMMIT_THRESHOLD);
+            Long txnId = suspendBig();
```
Comment (AaronCMuller) [inline src/amberdb/graph/AmberGraph.java]: I assume here that I shouldn't be surrounding this in a begin and commit with the dao.

Comment (scoen) [inline src/amberdb/graph/AmberGraph.java]: Yeah, for now your assumption is right. But i do need to revisit that in the near future - i think it's an issue i need to address.

File: src/amberdb/graph/AmberGraph.java [hunk 2]
```diff
@@ -250,7 +250,18 @@ public Vertex addVertex(Object id) {
...
+            log.debug("Commence clearing session");
+            dao.clearSession(txnId);
```
Comment (scoen) [inline src/amberdb/graph/AmberGraph.java]: i think this will delete the session that's just been written to the db

Comment (scoen) [inline src/amberdb/graph/AmberGraph.java]: i think maybe just clear() would clean up memory

Comment (AaronCMuller) [inline src/amberdb/graph/AmberGraph.java]: Oh yeah ... okay.

**Comment by shuangzhou** [comment]
I'll run a test of a large EAD over the change.

**Comment by brycollins** [comment]
Not in production Please.
Last time it caused a 15minute outage for all users.

**Comment by scoen** [comment]
Yes, a test run in devel would definitely be good thing

**Comment by shuangzhou** [comment]
I've tested suspend and commit against creation, reload of 6598 works in both session suspend and session commit, all worked.

**Pull request merged**